src/Security/DefaultAdminService.php

```diff
@@ -101,8 +162,19 @@ public static function hasDefaultAdmin()
     {
         // Check environment if not explicitly set
         if (!isset(static::$has_default_admin)) {
-            return !empty(Environment::getEnv('SS_DEFAULT_ADMIN_USERNAME'))
-                && !empty(Environment::getEnv('SS_DEFAULT_ADMIN_PASSWORD'));
+            $uniqueIdentifierFieldName = Member::config()->get('unique_identifier_field');
+            if($uniqueIdentifierFieldName != 'Email') {
+                return !empty(Environment::getEnv('SS_DEFAULT_ADMIN_' . strtoupper($uniqueIdentifierFieldName)))
```

Comment:
Why would all 3 of these need values to pass if the identifier field is not email?